### DISCUSSION TITLE
Error "module 'comfy.sample' has no attribute 'prepare_mask'" when executing KSampler Adv. (Efficient) in ComfyUI

### DISCUSSION THREAD

**Discussion by dalan1234** [Comfy-Org/ComfyUI#3234] [Q&A] [answered]
Hello everyone,

I've encountered an issue while working with ComfyUI, specifically when trying to execute the KSampler Adv. (Efficient) function. The error message I receive is:

javascript

Error occurred when executing KSampler Adv. (Efficient):
module 'comfy.sample' has no attribute 'prepare_mask'
This error occurs within my custom implementation based on the ComfyUI framework. Here are the relevant details of the error trace:

arduino

File "D:\comfy_ui\comfyui_blender\Blender_ComfyUI\ComfyUI\execution.py", line 151, in recursive_execute
    output_data, output_ui = get_output_data(obj, input_data_all)
...
File "D:\comfy_ui\comfyui_blender\Blender_ComfyUI\ComfyUI\custom_nodes\ComfyUI-AnimateDiff-Evolved\animatediff\sampling.py", line 217, in inject_functions
    self.orig_prepare_mask = comfy.sample.prepare_mask
It appears the prepare_mask attribute is missing from the comfy.sample module. I've checked my installation, and everything seems to be up to date.

I'm wondering if anyone has faced a similar issue or could provide insight into how this issue might be resolved. Here are a few additional details that might be helpful:

ComfyUI Version: (please specify your version)
Environment: (e.g., Blender plugin, specific version)
Operating System: (e.g., Windows 10)
Any help or guidance would be greatly appreciated as I'm currently stuck on this issue. Thank you in advance for your time and assistance!

**Comment by tianpcheg** [comment]
I encountered the same problem as you

**Reply by dalan1234** [reply, marked as answer]
我解决了，是采样器升级导致，可以暂时先不升级采样器，把animateDiff-evolve升级一下，我是删掉了以前的文件夹，重新装的

**Reply by tianpcheg** [reply]
我也升级了采样器，但是并没有解决这个问题

**Reply by al2h36o30s3** [reply]
可以问一下采样器是用的哪个版本的吗

**Comment by Kosinkadink** [comment]
You just need to update AnimateDiff-Evolved to the latest version. Comfy reached out to me before he pushed the update last week, so I had an update pushed out that would work with the newest ComfyUI before it even came out.

My code doesn't even reference prepare_mask at all anymore, so your version is definitely not updated. Also make sure you do not have any duplicate installs of AnimateDiff-Evolved in custom_nodes, and that AnimateLCM is not installed (it's just a copy of my code, not changed properly to not interfere, and is not maintained).

**Reply by tianpcheg** [reply]
Thanks, I successfully solved the problem

**Reply by ace2duce** [reply]
@Kosinkadink this save me as well, thanks 😊

**Reply by chieeo** [reply]
thanks

**Reply by philcoraz** [reply]
Yeah this folder AnimateLCM was the problem I deleted and it solved the problem thanks a lot

**Comment by hhh315xx** [comment]
thanks,it works

**Comment by uestcqxq** [comment]
thx, save me as well
